bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -83,44 +83,35 @@ def from_schema_type(schema_type):
         return None
 
 
-@attr.define(eq=True)
+@attrs.define(eq=True)
 class Column:
     """Representation of a column in a query, with relevant details for shredder mitigation."""
 
     name: str
-    data_type: DataTypeGroup = attr.field(default=DataTypeGroup.UNDETERMINED)
-    column_type: ColumnType = attr.field(default=ColumnType.UNDETERMINED)
-    status: ColumnStatus = attr.field(default=ColumnStatus.UNDETERMINED)
-
-    @data_type.validator
-    def validate_data_type(self, _attribute, value):
-        """Check that the type of data_type is as expected."""
-        if not isinstance(value, DataTypeGroup):
-            raise ValueError(f"Invalid {value} with type: {type(value)}.")
-
-    @column_type.validator
-    def validate_column_type(self, _attribute, value):
-        """Check that the type of parameter column_type is as expected."""
-        if not isinstance(value, ColumnType):
-            raise ValueError(f"Invalid data type for: {value}.")
-
-    @status.validator
-    def validate_status(self, _attribute, value):
-        """Check that the type of parameter column_status is as expected."""
-        if not isinstance(value, ColumnStatus):
-            raise ValueError(f"Invalid data type for: {value}.")
-
-
-@attr.define(eq=True)
+    data_type: DataTypeGroup = attrs.field(
+        default=DataTypeGroup.UNDETERMINED,
+        validator=attrs.validators.instance_of(DataTypeGroup),
+    )
+    column_type: ColumnType = attrs.field(
+        default=ColumnType.UNDETERMINED,
+        validator=attrs.validators.instance_of(ColumnType),
+    )
+    status: ColumnStatus = attrs.field(
+        default=ColumnStatus.UNDETERMINED,
+        validator=attrs.validators.instance_of(ColumnStatus),
+    )
+
+
+@attrs.define(eq=True)
 class Subset:
     """Representation of a subset/CTEs in the query and the actions related to this subset."""
 
     client: bigquery.Client
-    destination_table: str = attr.field(default="")
-    query_cte: str = attr.field(default="")
-    dataset: str = attr.field(default=TEMP_DATASET)
-    project_id: str = attr.field(default=DEFAULT_PROJECT_ID)
-    expiration_days: Optional[float] = attr.field(default=None)
+    destination_table: str = attrs.field(default="")
```

Comment:
Should default be specified here? Seems weird to set it to an empty string rather than throwing an error if it is required to execute the logic?

Reply:
Good call! Added a validator and test.